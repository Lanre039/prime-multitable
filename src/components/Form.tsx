import React from 'react';
import { IFormProps } from './types';
import utils from '../utils';
import './styles.css';

function Form(props: IFormProps): JSX.Element {
  const { formInput, setFormInput, setPrimeNo } = props;

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    setFormInput(value);
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const primeNo = utils.handlePrimeNo(formInput);
    if (primeNo) {
      setPrimeNo(primeNo);
    }
  };

  return (
    <section>
      <h1 className="head">PRIME MULTIPLICATION TABLE GENERATOR</h1>
      <form className="form" onSubmit={handleSubmit}>
        <div style={{ display: 'flex' }}>
          <input
            onChange={handleChange}
            value={formInput}
            name="formInput"
            placeholder="Enter positive integer..."
            required
          />
          <button type="submit" className="btn btn-2">
            Submit
          </button>
        </div>
      </form>
    </section>
  );
}

export default Form;
