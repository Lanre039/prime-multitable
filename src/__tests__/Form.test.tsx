import { render } from '@testing-library/react';
import Form from '../components/Form';

it('should take a snapshot', () => {
  const props = {
    formInput: '',
    setFormInput: jest.fn(),
    setPrimeNo: jest.fn(),
  };
  const { asFragment } = render(<Form {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
