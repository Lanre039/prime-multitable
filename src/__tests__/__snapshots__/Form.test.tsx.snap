// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      class="head"
    >
      PRIME MULTIPLICATION TABLE GENERATOR
    </h1>
    <form
      class="form"
    >
      <div
        style="display: flex;"
      >
        <input
          name="formInput"
          placeholder="Enter positive integer..."
          required=""
          value=""
        />
        <button
          class="btn btn-2"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</DocumentFragment>
`;
